LPS-199654, LPS-199653 | Test Automation | Master

davisaints opened the pull request:
JIRA Tickets: [LPS-199654](https://liferay.atlassian.net/browse/LPS-199654), [LPS-199653](https://liferay.atlassian.net/browse/LPS-199653)

chestofwonders committed:
LPS-196835 Improve icon selection in Item Actions creation page

Andrefes565 committed:
ISSD-1364 replace expando column

Andrefes565 committed:
ISSD-1364 replace blueprint

lfbesada committed:
LPS-200228 Prevent the content type changes when the display page is used by at least one asset

davisaints committed:
LPS-199654 Add bindToRootObject macro

davisaints committed:
LPS-199654 Add test to verify that the user can see the relationships tabs in the initial out-of-the-box layout of one level of children

davisaints committed:
LPS-199653 Add test to verify that published child object definitions do not have dedicated portlets and that the related configurations are read only

davisaints commented on modules/apps/object/object-test/src/testFunctional/paths/ObjectRootModel.path:
```diff
@@ -0,0 +1,21 @@
+
+<tbody>
+	<tr>
+		<td>DROPDOWN_MENU_WITH_INDEX</td>
```
Yes, I found generic paths for a dropdown menu, but to make [this macro](https://github.com/liferay-bpm-qa/liferay-portal/pull/1199/files#diff-f5e4d02c87df108d14fbfe54ab65ddc41ec9a5073871ff3879a35d78ee5f4c73R4), I needed a path that locates a dropdown menu using an index (we don't have) to make an increment [here](https://github.com/liferay-bpm-qa/liferay-portal/pull/1199/files#diff-f5e4d02c87df108d14fbfe54ab65ddc41ec9a5073871ff3879a35d78ee5f4c73R22)

davisaints commented on modules/apps/object/object-test/src/testFunctional/tests/ObjectRootModel.testcase:
```diff
@@ -0,0 +1,151 @@
+				fieldName = "textField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "Parent");
```
Since the child object already has a relationship field, we just need to create a field for the parent object

davisaints commented on modules/apps/object/object-test/src/testFunctional/tests/ObjectRootModel.testcase:
```diff
@@ -0,0 +1,151 @@
+
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "textField",
+				objectName = "SantaCruze",
```
It's not a typo. The plural label for this object in the API is SantaCruzes. As this macro adds an "s" to the value we give to objectName (See [here](https://github.com/davisaints/liferay-portal/blob/4f33caca1b181bc1a1a7968773ce7447f79449e7/modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro#L595)), I only needed to put SantaCruze.

veroglez committed:
LPS-200398 Add new vertical navigation for settings, and rename the existing one, to standardize namings

veroglez committed:
LPS-200398 Add different subtitle for structures

sandrodw3 committed:
LPS-200228 Not needed

sandrodw3 committed:
LPS-200228 Fix name

sandrodw3 committed:
LPS-200228 Rename and move to components folder

sandrodw3 committed:
LPS-200228 Rename to commands for consistency with other modules

sandrodw3 committed:
LPS-200228 Add openInUseModal command

sandrodw3 committed:
LPS-200228 Open in use modal if it initially has usages

evanilson-santana commented on modules/apps/object/object-test/src/testFunctional/macros/ObjectRootModel.macro:
```diff
@@ -0,0 +1,32 @@
+definition {
+
+	@summary = "Bind an object or multiple objects to the root object path"
```
I think this summary would be better for something like this: "Bind one or multiple object definitions to the root object." I don't know why you're using the word  "path".

davisaints replied:
Done [here](https://github.com/liferay-bpm-qa/liferay-portal/pull/1202/commits/953654685c974d6fcf6eb07ba1d90e9a4f1e6aac#diff-f5e4d02c87df108d14fbfe54ab65ddc41ec9a5073871ff3879a35d78ee5f4c73R3)

evanilson-santana commented on modules/apps/object/object-test/src/testFunctional/tests/ObjectRootModel.testcase:
```diff
@@ -21,6 +21,76 @@ definition {
 		}
 	}
 
+	@description = "LPS-189380 - Verify the user can to see relationships tabs in the initial out-of-the-box layout of one level of children"
```
You're duplicating the description here. It should be: _"Description LPS-189380 - Verify published child object definitions do not have dedicated portlets and the related configurations are read only"_  as stated in the ticket

davisaints replied:
Done [here](https://github.com/liferay-bpm-qa/liferay-portal/pull/1202/files#diff-03f0ca92d34d22561a353c49a77883f8a25588876d02cc52f3e4686e20d43188R24)

evanilson-santana reviewed (commented):
Hey @davisaints, good work here. It has only minor corrections. Tell me if you have any questions.

evanilson-santana commented on modules/apps/object/object-test/src/testFunctional/macros/ObjectRootModel.macro:
```diff
@@ -0,0 +1,32 @@
+definition {
+
+	@summary = "Bind an object or multiple objects to the root object path"
+	macro bindToRootObject(childObject,rootObjectPath) {
```
I don't know why you're using the word  "path".

davisaints replied:
This macro was made consistent with the interface. You can see this in the modal for binding objects:

![Objects-Objects-Objects-Liferay-DXP](https://github.com/liferay-bpm-qa/liferay-portal/assets/66960853/33baa1c3-d95d-47df-830d-a9e8e151bafd)

cc/ @timpak

veroglez committed:
LPS-200398 Adapt styles for vertical navigation titles

lfbesada committed:
LPS-200228 Handles ModelListenerException if the original cause is RequiredLayoutPageTemplateEntryException. We are handling this in the action command instead of the LayoutPageTemplateEntryExceptionRequestHandlerUtil since this error only should be processed in that way for this action command.

lfbesada committed:
LPS-200228 Adds other error properties needed by the front side to display the error dialog

lfbesada committed:
LPS-200228 Extract usageCount to reuse

lfbesada committed:
LPS-200228 SF use RenderURLBuilder

lfbesada committed:
LPS-200228 Pass viewUsagesURL instead of changeContentTypeURL when there are usages

lfbesada committed:
LPS-200228 Pass the usage count and use it in building the edit action dropdownItem.

Is a very unusual case that a DPT has some use but the type/subtype doesn't exist. I can only think of either bad data or the implementation of the type/subtype not being available at that time. That is why in this case, given that we cannot allow changing the type, we will simply disable the edition

lfbesada committed:
LPS-200228 SF cache the URLs to prevent calculating twice

chestofwonders committed:
LPS-196835 Refactor button group

sandrodw3 committed:
LPS-200228 Simplify default error

sandrodw3 committed:
LPS-200228 Consider hasUsages error

sandrodw3 committed:
LPS-200228 Improve other error handling and change display style

sandrodw3 committed:
LPS-200228 Change title and description when it has missing type

sandrodw3 committed:
LPS-200682 Select by default the current type and subtype

lfbesada committed:
LPS-200228 Add assetType, logic copied from DisplayPageVerticalCard no logic changes

lfbesada committed:
LPS-200682 Adds current classNameId and classTypeId data when opening the changeContentType form

lfbesada committed:
LPS-200228 Hide default success message in case handle the error

sandrodw3 committed:
LPS-200228 Types

sandrodw3 committed:
LPS-200682 Types

Andrefes565 committed:
ISSD-1364 formatSource

adolfopa committed:
LPS-199427 Export approved and scheduled articles

Stephy6 committed:
LPS-200964 Remove AUI related code and use native js methods instead in ratings.jsp in the portal-settings-web module

Stephy6 committed:
LPS-200964 Remove AUI related code and use native js methods instead in language.jsp in the portal-settings-web module

igor-franca committed:
LPS-197168 Remove Empty Node from Node Types

igor-franca committed:
LPS-197168 Create Empty Object Folder Card Component

p2kmgcl committed:
LPS-200398 Create HighlightedDDMStructuresConfiguration

jkappler committed:
LPS-200398 Move to display context

igor-franca committed:
LPS-201207 Increase and Standardize the Horizontal and Vertical Spaces Between Nodes in the Diagram

igor-franca committed:
LPS-201207 Store New Positions in ObjectFolderItem

igor-franca committed:
LPS-201208 Increment react flow edges z-index

markocikos committed:
LPS-196835 Enable input of icon symbol

markocikos committed:
LPS-196835 Tweak icon selection styles

Commit:
LPS-201343 put status in randomUserAccount()

jkappler committed:
LPS-200398 Read the preferences and return them to the view

Commit:
LPS-121479 Upgrade lodash dependency version to avoid vulnerability

Commit:
LPS-121479 Fix test failure

adolfopa committed:
LPS-199427 Don't assume article is approved

Schedule articles have status = STATUS_SCHEDULED.

liaoethan committed:
ISSD-118 remove quick links panel

liaoethan committed:
ISSD-118 remove language key

liaoethan committed:
ISSD-118 SF

JerryAdrianoo committed:
LPS-196835 Fix addActions according new changes

riccardo-ferrari committed:
LRAC-14977 analytics-client-js Update JS client

riccardo-ferrari committed:
LRAC-14977 journal-taglib Add resourcePrimaryKey to journal taglib

JerryAdrianoo committed:
LPS-196835 Fix CanCreateActionEntry test

JerryAdrianoo committed:
LPS-196835 Fix CannotSeeActionIconNotCreated test

JerryAdrianoo committed:
LPS-196835 Fix IconFieldIsNotMandatory test

JerryAdrianoo committed:
LPS-196835 Fix HeadlessActionKeyIsNotMandatory test

JerryAdrianoo committed:
LPS-196835 Fix LocalizedConfirmationMessageField test

JerryAdrianoo committed:
LPS-196835 Fix MessageTypeIsNotMandatory test

JerryAdrianoo committed:
LPS-196835 Fix IconFieldIsNotMandatory test

thaiscabrals committed:
LPS-200830 - Remove test related to DDL

liaoethan committed:
ISSD-118 remove lang keys for es, jp, and pt

michaelhashimoto committed:
LRCI-3823 Add a new queue to send messages from a webhook

michaelhashimoto committed:
LRCI-3850 Update activemq-client to 5.16.7

lfbesada committed:
LPS-200228 Adds required dependency

lfbesada committed:
LPS-200228 BuildLang

david-truong committed:
LPS-201365 use localService so that ctEntry gets indexed properly

jkappler committed:
LPS-200398 We already have a URL for item selector in display context, but the one that we need is more strict, so create a new one and specify in the criterion that we want to have multi select, and also show ancestor scopes based on configuration

Commit:
LPS-200398 Adds required attributes to the request

sandrodw3 committed:
LPS-201323 Add proper z-index to dropdown

Using 1021 because the problem comes from the management bar having sticky-top class, as it adds a z-index of 1020

rafaellajordao committed:
LPS-197895 add ViewLockedPagesAlphabeticalOrder test

rafaellajordao committed:
LPS-197895 add CanFilterAndSortViaKeyboard test

rafaellajordao committed:
LPS-197895 add ViewLockedPagesFilterAndOrderAccessibility test

rafaellajordao committed:
LPS-197895 add ViewLockedPagesFilterAndOrderScopePagesAlphabeticalOrder test

rafaellajordao committed:
LPS-201012 add Breadcrumb.gotoEntry macro

rafaellajordao committed:
LPS-201012 add NavigateByBreadcrumbInSearchResults test

Stephy6 committed:
LPS-200964 Remove AUI related code and use native js methods instead in view.jsp in the iframe-web module

rafaellajordao committed:
LPS-201012 add NavigateByBreadcrumbInSearchResultsViaKeyboard test

victorg1991 committed:
LPS-184267 Use lower case to make sure we compare the same values

victorg1991 committed:
LPS-201532 Use the published configuration as in other places

jkappler committed:
LPS-200398 Build lang

jkappler committed:
LPS-201247 Update test

Commit:
LPS-201012 Hardcode

kevenleone committed:
ISSD-1654 Move customer dashboard folder and content structure

pedroliveirav committed:
ISSD-1653 Refactor multiselect component and remove loop

Kvale1733 committed:
LRQA-84210 Assert script output if set

Kvale1733 committed:
LRQA-84210 Temp use UI to execute script

Commit:
LPS-201012 Add locator for list entry breadcrumb item

Commit:
LPS-201012 Simplify macro

Commit:
LPS-201012 Add macro to click list entry breadcrumb item

Commit:
LPS-201012 Refactor cases

davisaints committed:
LPS-200934 Fix UPLOAD_ON_WIDGET_ON_ANY_FIELD path

Commit:
LPS-197895 Remove ff

davisaints committed:
LPS-199654 Use bindToRootObject macro

Commit:
LPS-197895 Modify locator to make it reusable

Commit:
LPS-197895 Migrate and refactor cases

brianchandotcom committed:
LPS-196835 Wordsmith

brianchandotcom committed:
LPS-196835 Regen

brianchandotcom committed:
LRCI-3823 prep next

brianchandotcom committed:
LPS-200682 Wordsmith

brianchandotcom committed:
LPS-200682 SF

brianchandotcom committed:
LPS-200682 SF

Commit:
LPS-197895 Fix typo

Commit:
LPS-197895 Refactor cases

xbrianlee committed:
LPS-201592 Temporarily ignore - test needs to be reworked to fix timing issue

xbrianlee committed:
LPS-201232 Workaround

xbrianlee committed:
LPS-200635 Temporarily ignore due to broken scripts

Commit:
LPS-201552 Remove testcase and macro unnecessary

Commit:
LPS-201552 Refactor pending testcases related the clay framework changes

davisaints commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-31.liferay.com/job/test-portal-source-format/6001/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/6001/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 7 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/4fc9a18bd48612ada94911e0346b22239ed1dd81">4fc9a18bd48612ada94911e0346b22239ed1dd81</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/davisaints/liferay-portal/tree/LPS-199654">LPS-199654</a><br/>Branch GIT ID: <a href="https://github.com/davisaints/liferay-portal/commit/189e991ee0ee9d9f7366d84fcb4d3a30edffa1d3">189e991ee0ee9d9f7366d84fcb4d3a30edffa1d3</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-31.liferay.com/job/test-portal-source-format/6001/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/6001/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-31.liferay.com/job/test-portal-source-format/6001//consoleFull)

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/143814
[Console](https://test-1-31.liferay.com/job/test-portal-source-format/6001//consoleFull)

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-31.liferay.com/job/test-portal-source-format/6001/">test-portal-source-format#6001</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/6001/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-bpm-qa/liferay-portal/pull/1202">liferay-bpm-qa#1202</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2459228586">[master] ci:test:sf - davisaints &gt; liferay-bpm-qa - PR#1202 - 2023-11-14[11:42:21]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-31.liferay.com/job/publish-testray-report/22906/">publish-testray-report#22906</a><br/></div></div>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest(master)/9923/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/9923/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest(master)/9923/">test-portal-acceptance-pullrequest(master)#9923</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-31.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/9923/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-bpm-qa/liferay-portal/pull/1202">liferay-bpm-qa#1202</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2459241622">[master] ci:test:relevant - davisaints &gt; liferay-bpm-qa - PR#1202 - 2023-11-14[12:19:32]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-31.liferay.com/job/publish-testray-report/22908/">publish-testray-report#22908</a><br/></div></div>
